src/main/kotlin/io/ionic/libs/ionfilesystemlib/helper/common/IONFILECommon.kt

```diff
@@ -0,0 +1,167 @@
+package io.ionic.libs.ionfilesystemlib.helper.common
+
+import android.annotation.SuppressLint
+import android.net.Uri
+import android.os.Build
+import android.webkit.MimeTypeMap
+import io.ionic.libs.ionfilesystemlib.model.IONFILECreateOptions
+import io.ionic.libs.ionfilesystemlib.model.IONFILEDeleteOptions
+import io.ionic.libs.ionfilesystemlib.model.IONFILEExceptions
+import io.ionic.libs.ionfilesystemlib.model.IONFILEFileType
+import io.ionic.libs.ionfilesystemlib.model.IONFILEMetadataResult
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.attribute.BasicFileAttributes
+import kotlin.math.min
+
+// common methods and variables to different IONFILE helpers
+
+internal const val FILE_MIME_TYPE_FALLBACK = "application/octet-binary"
+
+/**
+ * Create a directory or file
+ *
+ * Most logic is common to both, except the actual creation
+ *
+ * @param fullPath full path to the file/directory to create
+ * @param options options to create the file/directory
+ * @param isDirectory true if meant to create directory, false if meant to create file
+ * @return success if the file/directory was created successfully, error otherwise
+ */
+internal fun createDirOrFile(
```

Comment:
This may be more a matter of opinion and it isn't super important but this could possibly be refactored to use the when clause instead of multiple if statements, like so:

    ```
val file = File(fullPath)
    when {
        file.exists() -> throw IONFILEExceptions.CreateFailed.AlreadyExists()
        !checkParentDirectory(file, create = options.recursive) -> throw IONFILEExceptions.CreateFailed.NoParentDirectory()
        isDirectory && !file.mkdir() -> throw IONFILEExceptions.UnknownError()
        !isDirectory && !file.createNewFile() -> throw IONFILEExceptions.UnknownError()
    }
```